### ISSUE TITLE
2.5.4 UNIT_ATTACK_SPEED doesn't trigger anymore

### ISSUE THREAD

**Issue by mrbuds** [Stanzilla/WoWUIBugs#231]
From various reports and my own testing it seems that the event `UNIT_ATTACK_SPEED` doesn't trigger anymore.

I have tested on a paladin using `Seal of the Crusader` that give 40% haste, and it was reported to me that there is same issue with haste potions, flurry for warriors and melee shamans spec.

Test code:

```Lua
local f = CreateFrame("FRAME")
f:RegisterUnitEvent("UNIT_ATTACK_SPEED", "player")
f:RegisterEvent("COMBAT_LOG_EVENT_UNFILTERED")
local myGUID = UnitGUID("player")

local function f_onEvent(self, event, ...)
   if event == "COMBAT_LOG_EVENT_UNFILTERED" then
      local timestamp, subEvent, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, spellId, spellName = CombatLogGetCurrentEventInfo()
      if sourceGUID == myGUID
      and spellName == "Seal of the Crusader" -- 40% haste
      then
         print(subEvent, GetTime(), UnitAttackSpeed("player"))
      end
   elseif event == "UNIT_ATTACK_SPEED" then
      print(event, GetTime(), UnitAttackSpeed("player"))
   end
end
f:SetScript("OnEvent", f_onEvent)
```

output when buffing/debuffing myself with seal of the crusader against a dangerous raptor
![image](https://user-images.githubusercontent.com/189656/160238980-2f474e7f-9a37-40a9-9e57-57b6edf3f079.png)

As you can see attack speed update on buff gain/lose (2.357 vs 3.30) but there is no `UNIT_ATTACK_SPEED` event

Test done on build 2.5.4.42917

**Comment by mrbuds** [comment]
not fixed in 2.5.4.42940

**Comment by Meorawr** [comment]
Appears to be fixed in 3.4.0 when testing with Flurry.
